covalent/_workflow/postprocessing.py

```diff
@@ -173,3 +174,66 @@ def add_exhaustive_postprocess_node(self, bound_electrons: Dict) -> None:
             )
             bound_pp = pp_electron(*filtered_ordered_electrons)
             tg.set_node_value(bound_pp.node_id, "name", postprocess_prefix)
+
+    def _postprocess_recursively(self, retval, **referenced_outputs):
+        from .electron import Electron
+
+        if isinstance(retval, Electron):
+            key = f"node:{retval.node_id}"
+            return referenced_outputs[key]
+        elif isinstance(retval, list):
+            return list(
+                map(lambda x: self._postprocess_recursively(x, **referenced_outputs), retval)
+            )
+        elif isinstance(retval, tuple):
+            return tuple(
+                map(lambda x: self._postprocess_recursively(x, **referenced_outputs), retval)
+            )
+        elif isinstance(retval, set):
+            return {self._postprocess_recursively(x, **referenced_outputs) for x in retval}
+        elif isinstance(retval, dict):
+            return {
+                k: self._postprocess_recursively(v, **referenced_outputs)
+                for k, v in retval.items()
+            }
+        else:
+            return retval
+
+    def add_reconstruct_postprocess_node(
+        self, retval: Union["Electron", List, Dict], bound_electrons: Dict
+    ):
+        """This function adds a postprocess node to the transport graph based on the 'eager reconstruction' algorithm.
+
+        Args:
+            retval: Return value of the lattice function.
+            bound_electrons: Dictionary of bound electrons with node_id as key and the electron object as the value. A bound is an electron that has been assigned a node_id and is part of a transport graph.
+
+        Returns:
+            None
+
+        """
+        from .electron import Electron, wait
+
+        node_id_refs = self._get_node_ids_from_retval(retval)
+        referenced_electrons = {}
+        for node_id in node_id_refs:
+            key = f"node:{node_id}"
+            referenced_electrons[key] = bound_electrons[node_id]
+
+        with active_lattice_manager.claim(self.lattice):
+            pp_metadata = self._get_electron_metadata()
+            pp_electron = Electron(function=self._postprocess_recursively, metadata=pp_metadata)
+
+            # Add pp_electron to the graph -- this will also add a
+            # parameter node in case retval is not a single electron
+            bound_pp = pp_electron(retval, **referenced_electrons)
+
+            # Edit pp electron name
+            self.lattice.transport_graph.set_node_value(
+                bound_pp.node_id, "name", f"{postprocess_prefix}reconstruct"
+            )
+
+            # Wait for non-referenced electrons
+            if get_config("sdk.reconstruct_postprocess") != "true":
```

Comment:
This isn't quite right -- the original condition was if an experimental `eager_postprocess` config option  was set, in which case postprocessing would kick off as soon as its dependencies were satisfied without waiting for other electrons. However that would cause a UX change as the workflow could be marked as "COMPLETED" before all electrons have finished executing (just the electrons in the connected component of the postprocessing electron).

Unless we want to expose such an "eager execution" option right now (@santoshkumarradha ), we should wait unconditionally for the non-referenced electrons so as to preserve the existing behavior in which postprocessing occurs as the last step in the workflow.

Reply:
I think "eager execution" would be mainly appropriate if users are constructing their workflow in a functional manner -- such that they only care about the workflow's final output. However if users are computing primarily with side effects, it would be wrong to declare a workflow as "COMPLETED" just because the workflow's output can be resolved.

Reply:
"Eager execution" would benefit users who construct their workflow in a functional manner -- such that they only care about the workflow's final output. However if users are computing primarily with side effects, it would be wrong to declare a workflow as "COMPLETED" just because the workflow's output can be resolved.